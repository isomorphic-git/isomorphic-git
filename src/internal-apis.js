import * as managers from './managers'
import * as models from './models'
import * as utils from './utils'
export { managers }
export { models }
export { utils }
export { listCommits } from './commands/push'
export { listObjects } from './commands/push'
export { pack } from './commands/push'
export { unpack } from './commands/fetch'
