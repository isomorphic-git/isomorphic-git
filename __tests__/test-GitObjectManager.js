/* global test describe expect */
import _fs from 'fs'
import { managers, models } from '../dist/for-node/internal-apis'
const { GitObjectManager } = managers
const { FileSystem } = models
const fs = new FileSystem(_fs)

describe('GitObjectManager', () => {
  test('test missing', async () => {
    let ref = GitObjectManager.read({
      fs,
      gitdir: '__tests__/__fixtures__/test-GitObjectManager.git',
      oid: 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa'
    })
    await expect(ref).rejects.toMatchSnapshot()
  })
  test('test shallow', async () => {
    let ref = GitObjectManager.read({
      fs,
      gitdir: '__tests__/__fixtures__/test-GitObjectManager.git',
      oid: 'b8b1fcecbc6f5ea8bc915c3ac319e8c9eb204f95'
    })
    await expect(ref).rejects.toMatchSnapshot()
  })
})
