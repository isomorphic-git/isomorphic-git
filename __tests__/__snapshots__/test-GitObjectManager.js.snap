// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitObjectManager test missing 1`] = `[ReadObjectFail: Failed to read git object with oid aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa]`;
